coffee-backend/src/main/kotlin/chat/letscoffee/meeting/MeetingService.kt

```diff
@@ -0,0 +1,46 @@
+package chat.letscoffee.meeting
+
+import chat.letscoffee.exception.ResourceNotFoundException
+import chat.letscoffee.exception.ServiceUnavailableException
+import chat.letscoffee.user.User
+import com.github.kittinunf.fuel.Fuel
+import com.github.kittinunf.fuel.gson.responseObject
+import org.springframework.stereotype.Component
+
+
+@Component
+class MeetingService(
+    private val meetingRepository: MeetingRepository
+) {
+
+    fun createMeetingRoom(user: User): Meeting {
+
+        val bodyJson = "{\"Title\" : \"Coffee With Me\"}"
+        val (request, response, result) = Fuel
+            .post("https://api.join.skype.com/v1/meetnow/createjoinlinkguest")
+            .header("Content-Type" to "application/json")
+            .body(bodyJson)
+            .responseObject<MeetingResponse>()
+
+        return result.fold(
+            {
+                var meeting = Meeting(
+                    owner = user,
+                    link = it.joinLink,
+                    provider = MeetingProviderType.SKYPE
+                )
+                meeting = meetingRepository.save(meeting)
+                return@fold meeting
+            },
+            { throw ServiceUnavailableException("Could not create Skype meeting room.", it) }
+        )
+    }
+
+    fun getMeetingRoomById(id: Long): Meeting {
+        return meetingRepository.findByIdAndActiveTrue(id) ?: throw ResourceNotFoundException("Meeting", "id", id)
+    }
+
+    fun update(meeting: Meeting): Meeting {
+        return meetingRepository.save(meeting)
```

Comment:
Should the controller be responsible for actually updating the object? Or should that be done here as well?

Reply:
Good point, should probably move that logic inside this function (and renaming it) 🦘